lib/kafka/consumer.rb

```diff
@@ -229,7 +231,10 @@ def each_message(min_bytes: 1, max_bytes: 10485760, max_wait_time: 1, automatica
             @instrumenter.instrument("process_message.consumer", notification) do
               begin
                 yield message
-                @current_offsets[message.topic][message.partition] = message.offset
+
+                unless paused?(message.topic, message.partition)
+                  @current_offsets[message.topic][message.partition] = message.offset
+                end
```

Comment:
I stepped through this issue as well and came to a very similar changeset.

Reply:
I think instead of doing this, we should break the loop for the entire batch. Let me take a look within the next few days.